src/sas/qtgui/Utilities/Preferences/PreferencesPanel.py

```diff
@@ -20,6 +20,7 @@
 BASE_PANELS = {"Plotting Settings": PlottingPreferencesWidget,
                "Display Settings": DisplayPreferencesWidget,
                }  # Type: Dict[str, Union[Type[PreferencesWidget], Callable[[],QWidget]]
+ConfigType = Union[str, bool, float, int, List[Union[str, float, int]]]
```

Comment:
Just a note, the full type of config elements can be defined as:

```
ConfigType = Union[List["ConfigType"], str, bool, float, int]
```

Reply:
...sort of, because as its finite, its actually:

```
ConfigTypeTmp = Union[int, float, str, bool]
ConfigSubtypes = []
for i in range(11):
    ConfigSubtypes.append(ConfigTypeTmp)
    ConfigTypeTmp = List[ConfigTypeTmp]
ConfigType = Union.__getitem__(tuple(ConfigSubtypes))
```

Reply:
Doesn't really matter for your purposes though

Reply:
Ah, yes, schema elements can be nested Lists. Theoretically, the nesting could be infinite, but practically speaking, never will be. Would it make more sense to put `ConfigType` into sas.system.config.config and import it from there?

Reply:
I put a hard limit on the nesting, because of `x = []`, `x.append(x)`. 

> Would it make more sense to put ConfigType into sas.system.config.config and import it from there?

Depends if you mean to represent the types that config can handle, or the types that the preferences can handle.